carina-api/src/main/java/com/qaprosoft/apitools/validation/JsonValidator.java

```diff
@@ -32,62 +37,98 @@
 import com.github.fge.jsonschema.main.JsonSchemaFactory;
 
 public class JsonValidator {
-    private final static Logger LOGGER = Logger.getLogger(JsonValidator.class);
+	private final static Logger LOGGER = Logger.getLogger(JsonValidator.class);
 
-    public static void validateJson(String expectedJson, String actualJson, JSONCompareMode jsonCompareMode) {
-        try {
-            JSONAssert.assertEquals(expectedJson, actualJson, new JsonKeywordsComparator(jsonCompareMode));
-        } catch (JSONException e) {
-            throw new RuntimeException(e);
-        }
-    }
+	public static void validateJson(String expectedJson, String actualJson, JSONCompareMode jsonCompareMode) {
+		try {
+			JSONAssert.assertEquals(expectedJson, actualJson, new JsonKeywordsComparator(jsonCompareMode));
+		} catch (JSONException e) {
+			throw new RuntimeException(e);
+		}
+	}
 
-    public static void validateJsonAgainstSchema(String jsonSchema, String jsonData) {
-        // create the Json nodes for schema and data
-        JsonNode schemaNode;
-        JsonNode data;
-        try {
-            schemaNode = JsonLoader.fromString(jsonSchema);
-        } catch (IOException e) {
-            throw new RuntimeException("Can't read schema from String: " + e.getMessage(), e);
-        }
-        try {
-            data = JsonLoader.fromString(jsonData);
-        } catch (IOException e) {
-            throw new RuntimeException("Can't read json from String: " + e.getMessage(), e);
-        }
+	public static void validateJsonAgainstSchema(String jsonSchema, String jsonData) {
+		if (jsonSchema.contains("http://json-schema.org/draft-03/schema")
+				|| jsonSchema.contains("http://json-schema.org/draft-04/schema")) {
+			LOGGER.info("JSON schema version draft-03/draft-04 was detected");
+			validateJsonAgainstSchemaV3V4(jsonSchema, jsonData);
+		} else {
+			LOGGER.info("JSON schema version higher than draft-03/draft-04 was detected");
```

Comment:
seems like message is invalid (really draft-03/04 in this use-case?)

Reply:
this message is correct. we detect schemas of version 03/04 and use old validation logic for them. for other versions newly implemented logic is used